src/components/withCurrentUserPersonalDetails.tsx

```diff
@@ -0,0 +1,67 @@
+import React, {ComponentType, RefAttributes, ForwardedRef, useMemo} from 'react';
+import {OnyxEntry, withOnyx} from 'react-native-onyx';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+import ONYXKEYS from '../ONYXKEYS';
+import personalDetailsPropType from '../pages/personalDetailsPropType';
+import type {PersonalDetails, Session} from '../types/onyx';
+
+type CurrentUserPersonalDetails = PersonalDetails | Record<string, never>;
+
+type OnyxProps = {
+    /** Personal details of all the users, including current user */
+    personalDetails: OnyxEntry<Record<string, PersonalDetails>>;
+
+    /** Session of the current user */
+    session: OnyxEntry<Session>;
+};
+
+type HOCProps = {
+    currentUserPersonalDetails: CurrentUserPersonalDetails;
+};
+
+type ComponentProps = OnyxProps & HOCProps;
+
+// TODO: remove when all components that use it will be migrated to TS
+const withCurrentUserPersonalDetailsPropTypes = {
+    currentUserPersonalDetails: personalDetailsPropType,
+};
+
+const withCurrentUserPersonalDetailsDefaultProps: HOCProps = {
+    currentUserPersonalDetails: {},
+};
+
+export default function <TProps extends ComponentProps, TRef>(
+    WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>,
+): ComponentType<Omit<Omit<TProps, keyof HOCProps> & RefAttributes<TRef>, keyof OnyxProps>> {
+    function WithCurrentUserPersonalDetails(props: Omit<TProps, keyof HOCProps>, ref: ForwardedRef<TRef>) {
+        const accountID = props.session?.accountID ?? 0;
+        const accountPersonalDetails = props.personalDetails?.[accountID];
```

Comment:
Just one question - can these be null (i.e. do they need the `?`s? Seems like they'll always be there, right? Not marked as optional in OnyxProps...

Reply:
@dangrous Since `session` and `personalDetails` are OnyxEntry, and [OnyxEntry can be null](https://github.com/Expensify/react-native-onyx/blob/main/lib/types.d.ts#L154), we need the `?`s.